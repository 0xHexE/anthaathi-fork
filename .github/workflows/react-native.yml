name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
