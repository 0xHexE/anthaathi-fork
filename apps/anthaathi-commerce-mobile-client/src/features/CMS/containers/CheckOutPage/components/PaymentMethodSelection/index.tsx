import React, {useState} from 'react';
import {Pressable, View} from 'react-native';
import {Card, Divider, Text} from 'react-native-paper';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {CheckOutPageComponentType} from '../../../../types/common';

type PaymentOptions = {
  key: string;
  name: string;
};

export interface PaymentMethodProps {
  title: string;
  options: PaymentOptions[];
}

const PaymentMethodSelection = (props: PaymentMethodProps) => {
  const [checkedKey, setCheckedKey] = useState('');
  return (
    <View style={{marginHorizontal: 10}} testID="paymentMethodSelection">
      <Text style={{color: '#364A15', fontSize: 16, fontWeight: '600'}}>
        {props.title}
      </Text>

      <Card
        style={{
          borderColor: '#E3E2E7',
          borderWidth: 1,
          borderRadius: 4,
          marginVertical: 5,
        }}>
        <Card.Content>
          {props.options.map(item => {
            return (
              <OptionItem
                key={item.key}
                item={item}
                checkedKey={checkedKey}
                setCheckedKey={setCheckedKey}
              />
            );
          })}
        </Card.Content>
      </Card>
    </View>
  );
};

const OptionItem = ({
  item,
  checkedKey,
  setCheckedKey,
}: {
  item: PaymentOptions;
  checkedKey: string;
  setCheckedKey: React.Dispatch<React.SetStateAction<string>>;
}) => {
  return (
    <View>
      <Pressable
        onPress={() => {
          setCheckedKey(item.key);
        }}>
        <View
          style={{
            marginVertical: 7,
            flexDirection: 'row',
          }}>
          <Icon
            name={item.key === checkedKey ? 'circle-slice-8' : 'circle-outline'}
            color="#008D3E"
            size={18}
          />
          <Text
            style={{
              marginLeft: 10,
              color: '#364A15',
              fontSize: 14,
              fontWeight: item.key === checkedKey ? '600' : '400',
            }}>
            {item.name}
          </Text>
        </View>
      </Pressable>
      <Divider style={{backgroundColor: '#E3E2E7', marginVertical: 3}} />
    </View>
  );
};

export default PaymentMethodSelection;

export const PaymentMethodSelectionCMSInput = {
  _component: CheckOutPageComponentType.PaymentMethodSelection,
  component: PaymentMethodSelection,
};
