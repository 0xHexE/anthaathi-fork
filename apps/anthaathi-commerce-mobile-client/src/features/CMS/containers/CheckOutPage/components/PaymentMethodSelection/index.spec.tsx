import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import PaymentMethodSelection from './index';

describe('PaymentMethodSelection', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <PaymentMethodSelection
            title="Payment Method"
            options={[
              {
                key: '1',
                name: 'Credit / Debit Card',
              },
              {
                key: '2',
                name: 'Cash on delivery',
              },
              {
                key: '3',
                name: 'Wallet',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('paymentMethodSelection')).toBeTruthy();
  });

  it('should have PaymentMethodSelection title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <PaymentMethodSelection
            title="Payment Method"
            options={[
              {
                key: '1',
                name: 'Credit / Debit Card',
              },
              {
                key: '2',
                name: 'Cash on delivery',
              },
              {
                key: '3',
                name: 'Wallet',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('Payment Method')).toBeTruthy();
  });
});
