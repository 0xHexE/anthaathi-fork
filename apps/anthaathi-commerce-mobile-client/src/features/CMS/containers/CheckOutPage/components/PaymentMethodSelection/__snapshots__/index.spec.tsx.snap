// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentMethodSelection should render the item 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 10,
    }
  }
  testID="paymentMethodSelection"
>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "color": "#364A15",
          "fontSize": 16,
          "fontWeight": "600",
        },
      ]
    }
  >
    Payment Method
  </Text>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 3,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 1,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(255, 251, 254, 1)",
            "borderColor": "#E3E2E7",
            "borderRadius": 4,
            "borderWidth": 1,
            "elevation": 1,
            "marginVertical": 5,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexGrow": 1,
              "flexShrink": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "paddingHorizontal": 16,
                },
                Object {
                  "paddingVertical": 16,
                },
                undefined,
              ]
            }
          >
            <View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginVertical": 7,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#008D3E",
                          "fontSize": 18,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    󰝦
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "#364A15",
                          "fontSize": 14,
                          "fontWeight": "400",
                          "marginLeft": 10,
                        },
                      ]
                    }
                  >
                    Credit / Debit Card
                  </Text>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgba(231, 224, 236, 1)",
                      "height": 0.5,
                    },
                    undefined,
                    false,
                    false,
                    Object {
                      "backgroundColor": "#E3E2E7",
                      "marginVertical": 3,
                    },
                  ]
                }
              />
            </View>
            <View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginVertical": 7,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#008D3E",
                          "fontSize": 18,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    󰝦
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "#364A15",
                          "fontSize": 14,
                          "fontWeight": "400",
                          "marginLeft": 10,
                        },
                      ]
                    }
                  >
                    Cash on delivery
                  </Text>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgba(231, 224, 236, 1)",
                      "height": 0.5,
                    },
                    undefined,
                    false,
                    false,
                    Object {
                      "backgroundColor": "#E3E2E7",
                      "marginVertical": 3,
                    },
                  ]
                }
              />
            </View>
            <View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginVertical": 7,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#008D3E",
                          "fontSize": 18,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    󰝦
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "#364A15",
                          "fontSize": 14,
                          "fontWeight": "400",
                          "marginLeft": 10,
                        },
                      ]
                    }
                  >
                    Wallet
                  </Text>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgba(231, 224, 236, 1)",
                      "height": 0.5,
                    },
                    undefined,
                    false,
                    false,
                    Object {
                      "backgroundColor": "#E3E2E7",
                      "marginVertical": 3,
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
